aiidalab_widgets_base/codes.py

```diff
@@ -78,38 +73,26 @@ def __init__(self,
     def _get_codes(self):
         """Query the list of available codes."""
 
-        querybuild = QueryBuilder()
-        querybuild.append(Code,
-                          filters={
-                              'attributes.input_plugin': {
-                                  '==': self.input_plugin
-                              },
-                              'extras.hidden': {
-                                  "~==": True
-                              }
-                          },
-                          project=['*'])
-
-        # Whether to show or not hidden codes.
-        if self.hidden_codes:
-            to_return = [c[0] for c in querybuild.all()]
-        else:
-            to_return = [c[0] for c in querybuild.all() if not c[0].hidden]
-
-        # Whether to show the codes on disabled computers.
-        to_return = to_return if self.disabled_computers else [
-            c for c in to_return if c.computer.is_user_enabled(User.objects.get_default())
-        ]
+        user = User.objects.get_default()
 
-        # Only codes on computers configured for the current user.
         return {
-            self._full_code_label(c): c for c in to_return if c.computer.is_user_configured(User.objects.get_default())
+            self._full_code_label(c[0]): c[0]
+            for c in QueryBuilder().append(Code, filters={
+                'attributes.input_plugin': self.input_plugin
+            }).all()
+            if c[0].computer.is_user_configured(user) and (self.allow_hidden_codes or not c[0].hidden) and
+            (self.allow_disabled_computers or c[0].computer.is_user_enabled(user))
         }
 
     @staticmethod
     def _full_code_label(code):
         return "{}@{}".format(code.label, code.computer.name)
 
+    @observe('allow_disabled_computers')
+    def _observe_allow_disabl_comp(self, _=None):
```

Comment:
```suggestion
    def _observe_allow_disabled_computers(self, _=None):
```
Better to be consistent.

Reply:
ok